File: iosApp/iosApp/Presentation/Main/CafeManagement/MenuGoodsEdit/MenuGoodsEditAction.swift
//
//  MenuGoodsEditAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/11.
//

import Foundation

enum MenuGoodsEditAction {
    case clickBack
    case clickPhotoUpload
    case changeName(String)
    case changePrice(String)
    case selectPhoto(String)
    case selectCategory(String)
    case changeDescription(String)
    case toggleStock(Bool)
    case clickSave
    case dismissInfoMessage
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/cafemanagement/menugoodsedit/MenuGoodsEditAction.kt
package com.hhp227.concafe.presentation.main.cafemanagement.menugoodsedit

sealed interface MenuGoodsEditAction {
    data object ClickBack : MenuGoodsEditAction
    data object ClickPhotoUpload : MenuGoodsEditAction
    data class ChangeName(val value: String) : MenuGoodsEditAction
    data class ChangePrice(val value: String) : MenuGoodsEditAction
    data class SelectPhoto(val imageUrl: String) : MenuGoodsEditAction
    data class SelectCategory(val categoryId: String) : MenuGoodsEditAction
    data class ChangeDescription(val value: String) : MenuGoodsEditAction
    data class ToggleStock(val isInStock: Boolean) : MenuGoodsEditAction
    data object ClickSave : MenuGoodsEditAction
    data object DismissInfoMessage : MenuGoodsEditAction
}
